pkg/cmd/docs-issue-generation/docs_issue_generation.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type commit struct {
+	sha         string
+	title       string
+	releaseNote string
+}
+
+type pr struct {
+	number      int
+	mergeBranch string
+	commits     []commit
+}
+
+type ghSearch struct { // this struct holds the search results of all PRs based on a given commit
+	Items []ghSearchItem `json:"items"`
+}
+
+type ghSearchItem struct {
+	Number      int            `json:"number"` // PR number
+	PullRequest ghSearchItemPr `json:"pull_request"`
+}
+
+type ghSearchItemPr struct { // details about the pull request
```

Comment:
Same comment as with functions: it is appropriate to put the comment describing the `struct` above the `struct` definition.

```
// ghSearchItemPr contains details about the pull request.
type ghSearchItemPr struct {
...
```

Reply:
Done.